actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1935,6 +1935,11 @@ def path_scope(path)
           end
 
           def map_match(paths, options)
+            ActionDispatch.deprecator.warn(<<-MSG.squish) if paths.count > 1
+              Mapping a route with multiple paths is deprecated and
+              will be removed in Rails 8.0. Please use multiple method calls instead.
```

Comment:
I think `main` is already Rails 8. Is this patch intended to 7.x?

Reply:
Good question. This is a deprecation, so it could target 7.x, but since 7.2 is already in beta I think you're right.  I'll target 8.0 instead, and remove in 8.1. 👍